### DISCUSSION TITLE
reset optimizer for multiple runs

### DISCUSSION THREAD

**Discussion by aRI0U** [Lightning-AI/pytorch-lightning#7569] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,
I need to run multiple similar trainings in order to evaluate the average performance of my model, so I need to reset the weights of my network as well as the state of my optimizer, scheduler, etc.
I use the same `LightningModule` for all runs but a new `Trainer` for each run. I manually reset the weights of the network between each run so there is no issue here, however I don't know if the optimizers are reset when I pass the `LightningModule` to a new `Trainer`. If not, does anyone know how to reset them? It seems that I can access the optimizer with `self.optimizers().optimizer` but I don't know to what extent I'm able to overwrite it easily.
Thanks a lot!

**Comment by carmocca** [comment]
> I don't know if the optimizers are reset when I pass the LightningModule to a new Trainer

It's not that we reset them manually, `configure_optimizers` is called again in the next `fit` call.

> but I don't know to what extent I'm able to overwrite it easily.

You should be able to modify their state

**Reply by aRI0U** [reply]
Ok so basically each time I call `fit` new optimizers are created right?

**Reply by carmocca** [reply]
yes
